source/daplink/drag-n-drop/flash_manager.c

```diff
@@ -99,9 +113,34 @@ error_t flash_manager_init(const flash_intf_t *flash_intf)
 
     if (!page_erase_enabled) {   
         // Erase flash and unint if there are errors
+#if defined(TARGET_MCU_PSOC6) && defined(DAPLINK_IF)
+				flash_region_t region = FLASH_REGION_SFLASH_TOC2;
+			  if (kit_has_sflash_restriction())
+				{
+					region = FLASH_REGION_WFLASH;
+				}
+				while ( 1 ) 
+				{
+					status = switch_target_flash_algo(region);
+					if ( status != ERROR_SUCCESS )
+						break;
+					status = flash_intf_target->erase_chip();
+					if ( status != ERROR_SUCCESS && region != FLASH_REGION_SMIF )
+						break;
+					flash_manager_printf("    intf->erase_chip[%i] ret=%i\r\n", region, status);
+					if ( region != FLASH_REGION_MAIN )
+					{
+						region--;
+					}
+					else
+					{
+						break;
+					}
+				} 
+#else
```

Comment:
Currently there is no api to handle different regions but the target flash algo can be modified to insert this logic?

Reply:
Please take a look at #612 to eliminate this logic

Reply:
this logic has been eliminated